### PR TITLE
Add Search Bar to CustomNavBar

### PR DESCRIPTION
**Pull request by fvonhoven**
Adds a Search Bar & more functionality to the`CustomNavBar` as well as a way to search a ListView
(Also a few refactors & cleanups)
## Please verify the following:
- [X] Everything works on iOS/Android
- [X] `ignite-base` **ava** tests pass
- [X] `fireDrill.sh` passed

#### iOS
ignite - search

#### Android
ignite - search-android

### CODE REVIEW

**Comment by skellock** [comment]
Any chance you could rebase this?  It's hard to see what's changed.

**Comment by GantMan** [comment]
LGTM 👍 

Do you think we should do a generator for this option?

File: ignite-base/App/Config/ReduxPersist.js [hunk 1]
```diff
@@ -3,10 +3,10 @@ import { AsyncStorage } from 'react-native'
 
 const REDUX_PERSIST = {
   active: true,
-  reducerVersion: '2',
+  reducerVersion: '4',
   storeConfig: {
     storage: AsyncStorage,
-    blacklist: ['login'], // reducer keys that you do NOT want stored to persistence here
+    blacklist: ['login', 'search'], // reducer keys that you do NOT want stored to persistence here
```
Comment (skellock) [inline ignite-base/App/Config/ReduxPersist.js]: I'm thinking we should switch to preferring `whitelist`.   Not relevant to this PR, but just thinking outloud.

File: ignite-base/App/Containers/ListviewSearchingExample.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    if (newProps.results) {
+      this.setState({
+        dataSource: this.state.dataSource.cloneWithRows(newProps.results)
+      })
+    }
+  }
+
+  // Used for friendly AlertMessage
+  // returns true if the dataSource is empty
+  _noRowData () {
```
Comment (skellock) [inline ignite-base/App/Containers/ListviewSearchingExample.js]: I'm not a fan of `_`.  :(    Again, just general bitching.  Nothing really do with this PR.  Hah... oh boy.  I'm annoying myself now.

File: ignite-base/App/Config/ReduxPersist.js (hunk 1 above)
Comment (GantMan) [inline ignite-base/App/Config/ReduxPersist.js]: I generally prefer blacklisting in most apps.  We should discuss.

File: ignite-base/App/Containers/ListviewSearchingExample.js (hunk 2 above)
Comment (GantMan) [inline ignite-base/App/Containers/ListviewSearchingExample.js]: how should we indicate private methods?  If you have a better way, I'm in.

**Comment by GantMan** [comment]
yay

**Pull request merged**